Sources/Basic/TemporaryFile.swift

```diff
@@ -45,18 +46,13 @@ private extension TempFileError {
 /// - Returns: Path to directory in which temporary file should be created.
 public func determineTempDirectory(_ dir: AbsolutePath? = nil) throws -> AbsolutePath {
     // FIXME: Add other platform specific locations.
-    let tmpDir = dir ?? cachedTempDirectory
+    let tmpDir = dir ?? AbsolutePath(NSTemporaryDirectory())
```

Comment:
Does NSTemporaryDirectory look at those env variables?

Reply:
I believe that it does look at the proper variable for Darwin and Linux, not on Windows and I don't think it does so on Android.

Reply:
@millenomi can you confirm?

Reply:
On Windows, it calls into GetTempPathW which will look in:
- The path specified by the TMP environment variable.
- The path specified by the TEMP environment variable.
- The path specified by the USERPROFILE environment variable.
- The Windows directory.

On Darwin, it checks confstr for _CS_DARWIN_USER_TEMP_DIR 
On Linux or if it can't get _CS_DARWIN_USER_TMP_DIR it uses TMPDIR, and falls back to /tmp

On Android, it it falls back to /data/local/tmp

https://github.com/apple/swift-corelibs-foundation/blob/master/Foundation/NSPathUtilities.swift#L13

Reply:
I am a little bit worried that we'll change the behavior on Darwin but `NSTemporaryDirectory` is supposed to be the correct way to do it anyway.